LayoutKit/Views/ReloadableViewLayoutAdapter.swift

```diff
@@ -44,230 +44,129 @@ public class ReloadableViewLayoutAdapter: NSObject {
      */
     public func reload<T: CollectionType, U: CollectionType where U.Generator.Element == Layout, T.Generator.Element == Section<U>>(
-        width width: CGFloat = CGFloat.max,
-              height: CGFloat = CGFloat.max,
+        width width: CGFloat? = nil,
+              height: CGFloat? = nil,
               synchronous: Bool = false,
+              batchUpdates: BatchUpdates? = nil,
               layoutProvider: Void -> T,
-              completion: (Void -> Void)? = nil) {
+              completion: (() -> Void)? = nil) {
 
         assert(NSThread.isMainThread(), "reload must be called on the main thread")
 
         // All previous layouts are invalid.
         backgroundLayoutQueue.cancelAllOperations()
 
-        guard width > 0 && height > 0 else {
-            return
-        }
-
-        guard let reloadableView = reloadableView else {
-            return
-        }
-
-        let axis = reloadableView.scrollAxis() // avoid capturing the reloadableView in the layout function.
-        func layout(layout: Layout) -> LayoutArrangement {
-            switch axis {
-            case .vertical:
-                return layout.arrangement(width: width)
-            case .horizontal:
-                return layout.arrangement(height: height)
-            }
+        let layoutFunc = { (layout: Layout) -> LayoutArrangement in
+            return layout.arrangement(width: width, height: height)
         }
 
         if synchronous {
-            reloadSynchronously(layoutFunc: layout, layoutProvider: layoutProvider, completion: completion)
+            reloadSynchronously(with: layoutProvider, layoutFunc: layoutFunc, batchUpdates: batchUpdates, completion: completion)
         } else {
-            reloadAsynchronously(layoutFunc: layout, layoutProvider: layoutProvider, completion: completion)
+            reloadAsynchronously(with: layoutProvider, layoutFunc: layoutFunc, batchUpdates: batchUpdates, completion: completion)
         }
     }
 
-
-    /**
-     Reloads the view with a precomputed layout.
-     It must be called on the main thread.
-
-     This is useful if you want to precompute the layout for this collection view as part of another layout.
-     One example is nested collection/table views (see NestedCollectionViewController.swift in the sample app).
-     */
-    public func reload(arrangement arrangement: [Section<[LayoutArrangement]>]) {
-        assert(NSThread.isMainThread(), "reload must be called on the main thread")
-        backgroundLayoutQueue.cancelAllOperations()
-        currentArrangement = arrangement
-        reloadableView?.reloadDataSync()
-    }
-
     private func reloadSynchronously<T: CollectionType, U: CollectionType where U.Generator.Element == Layout, T.Generator.Element == Section<U>>(
-        layoutFunc layoutFunc: Layout -> LayoutArrangement,
-                   layoutProvider: Void -> T,
-                   completion: (Void -> Void)? = nil) {
+        with layoutProvider: Void -> T,
```

Comment:
is with the right word here? It seems to add no additional meaning?

`reloadSynchronously(with: provider, layoutFunc: func...)`
vs
`reloadSynchronously(provider, layoutFunc: func...)`
vs
`reloadSynchronously(layoutProvider: provider, layoutFunc: func...)`

It seems like 1 and 2 are mostly equivalent. 3 gives more info and is probably what I'd go for?